test/mlnet.Tests/TrainerGeneratorTests.cs

```diff
@@ -1,17 +1,41 @@
 ﻿using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
 using Microsoft.ML;
 using Microsoft.ML.AutoML;
 using Microsoft.ML.CLI.CodeGenerator.CSharp;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 
 namespace mlnet.Tests
 {
-    /****************************
-     * TODO : Add all trainer tests :
-     * **************************/
     [TestClass]
     public class TrainerGeneratorTests
     {
+        [TestMethod]
+        public void CultureInvariantTest()
+        {
+
+            var context = new MLContext();
+
+            var elementProperties = new Dictionary<string, object>()
+            {
+                {"LearningRate", 0.1f },
+                {"NumberOfLeaves", 1 },
+            };
+            PipelineNode node = new PipelineNode("LightGbmBinary", PipelineNodeType.Trainer, default(string[]), default(string), elementProperties);
+            Pipeline pipeline = new Pipeline(new PipelineNode[] { node });
+
+            //Set culture to deutsch.
+            Thread.CurrentThread.CurrentCulture = new CultureInfo("de-DE");
```

Comment:
Do we want to set back to normal afterwards? Or does this thread get disposed?

Reply:
The tests are run as task =>  a thread from the thread pool . That particular executing thread will have that culture set.
The culture needs to be reset again I guess. Though it doesn't have any side effects now. Will send out another PR for that